packages/expo-image/ios/EXImageView.m

```diff
@@ -23,12 +46,31 @@ - (void)dealloc
 
 - (void)setSource:(NSDictionary *)source
 {
-  _imageURL = [RCTConvert NSURL:source[sourceUriKey]];
+  _source = source;
+  // TODO: Implement equatable image source abstraction
+  _needsReload = YES;
+}
+
+- (void)setResizeMode:(RCTResizeMode)resizeMode
+{
+  if (_resizeMode == resizeMode) {
+    return;
+  }
+  
+  // Image needs to be reloaded whenever repeat is enabled or disabled
+  _needsReload = _needsReload || (resizeMode == RCTResizeModeRepeat) || (_resizeMode == RCTResizeModeRepeat);
+  _resizeMode = resizeMode;
+  
+  // Repeat resize mode is handled by the UIImage. Use scale to fill
+  // so the repeated image fills the UIImageView.
+  self.contentMode = resizeMode == RCTResizeModeRepeat
+    ? UIViewContentModeScaleToFill
+    : (UIViewContentMode)resizeMode;
 }
 
 - (void)didSetProps:(NSArray<NSString *> *)changedProps
 {
-  if ([changedProps containsObject:@"source"]) {
+  if (_needsReload) {
```

Comment:
We never set this to `NO` (apart from the initializer).

Reply:
Yup I noticed that as well. Fixed now!